From 2014de8cd3d20a7626e6cca157f522069a3628d0 Mon Sep 17 00:00:00 2001
From: Lei wang <leiwan@codeaurora.org>
Date: Mon, 19 Jul 2021 13:30:11 +0800
Subject: [PATCH] lttng-module: fix lttng-module compile issue

Android kernel is a little different from linux standard kernel,
this may cause lttng compile issue.
Make lttng-modules use new kallsyms_lookup_name.
Import android GKI symbol namespace.

Upstream-Status: Inappropriate [other]

Signed-off-by: Lei wang <leiwan@codeaurora.org>
---
 wrapper/kallsyms.c | 4 ++--
 wrapper/kallsyms.h | 3 ++-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/wrapper/kallsyms.c b/wrapper/kallsyms.c
index 6af77f5..6c100a4 100644
--- a/wrapper/kallsyms.c
+++ b/wrapper/kallsyms.c
@@ -14,10 +14,10 @@
 #include <linux/module.h>
 #include <wrapper/kallsyms.h>
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5,7,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5,4,0))
 
 #ifndef CONFIG_KPROBES
-# error "LTTng-modules requires CONFIG_KPROBES on kernels >= 5.7.0"
+# error "LTTng-modules requires CONFIG_KPROBES on kernels >= 5.4.0"
 #endif
 
 static
diff --git a/wrapper/kallsyms.h b/wrapper/kallsyms.h
index 5f48847..d54a084 100644
--- a/wrapper/kallsyms.h
+++ b/wrapper/kallsyms.h
@@ -26,8 +26,9 @@
 # endif
 #endif
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5,7,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5,4,0))
 
+MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
 unsigned long wrapper_kallsyms_lookup_name(const char *name);
 
 #else /* #if (LINUX_VERSION_CODE >= KERNEL_VERSION(5,7,0)) */
-- 
2.7.4

